src/object/schema_system_catalog_install.cpp

```diff
@@ -0,0 +1,1972 @@
+  ADD_TABLE_DEFINITION (CT_SERIAL_NAME, system_catalog_initializer::get_serial());
+  ADD_TABLE_DEFINITION (CT_HA_APPLY_INFO_NAME, system_catalog_initializer::get_ha_apply_info());
+  ADD_TABLE_DEFINITION (CT_COLLATION_NAME, system_catalog_initializer::get_collations());
+  ADD_TABLE_DEFINITION (CT_CHARSET_NAME, system_catalog_initializer::get_charsets());
+  ADD_TABLE_DEFINITION (CT_DUAL_NAME, system_catalog_initializer::get_dual());
+  ADD_TABLE_DEFINITION (CT_SYNONYM_NAME, system_catalog_initializer::get_synonym());
+  ADD_TABLE_DEFINITION (CT_DB_SERVER_NAME, system_catalog_initializer::get_db_server());
+
+  ADD_VIEW_DEFINITION (CTV_CLASS_NAME, system_catalog_initializer::get_view_class ());
+  ADD_VIEW_DEFINITION (CTV_SUPER_CLASS_NAME, system_catalog_initializer::get_view_super_class ());
+  ADD_VIEW_DEFINITION (CTV_VCLASS_NAME, system_catalog_initializer::get_view_vclass ());
+  ADD_VIEW_DEFINITION (CTV_ATTRIBUTE_NAME, system_catalog_initializer::get_view_attribute ());
+  ADD_VIEW_DEFINITION (CTV_ATTR_SD_NAME, system_catalog_initializer::get_view_attribute_set_domain ());
+  ADD_VIEW_DEFINITION (CTV_METHOD_NAME, system_catalog_initializer::get_view_method ());
+  ADD_VIEW_DEFINITION (CTV_METHARG_NAME, system_catalog_initializer::get_view_method_argument ());
+  ADD_VIEW_DEFINITION (CTV_METHARG_SD_NAME, system_catalog_initializer::get_view_method_argument_set_domain ());
+  ADD_VIEW_DEFINITION (CTV_METHFILE_NAME, system_catalog_initializer::get_view_method_file ());
+  ADD_VIEW_DEFINITION (CTV_INDEX_NAME, system_catalog_initializer::get_view_index ());
+  ADD_VIEW_DEFINITION (CTV_INDEXKEY_NAME, system_catalog_initializer::get_view_index_key ());
+  ADD_VIEW_DEFINITION (CTV_AUTH_NAME, system_catalog_initializer::get_view_authorization ());
+  ADD_VIEW_DEFINITION (CTV_TRIGGER_NAME, system_catalog_initializer::get_view_trigger ());
+  ADD_VIEW_DEFINITION (CTV_PARTITION_NAME, system_catalog_initializer::get_view_partition ());
+  ADD_VIEW_DEFINITION (CTV_STORED_PROC_NAME, system_catalog_initializer::get_view_stored_procedure ());
+  ADD_VIEW_DEFINITION (CTV_STORED_PROC_ARGS_NAME, system_catalog_initializer::get_view_stored_procedure_arguments ());
+  ADD_VIEW_DEFINITION (CTV_DB_COLLATION_NAME, system_catalog_initializer::get_view_db_collation ());
+  ADD_VIEW_DEFINITION (CTV_DB_CHARSET_NAME, system_catalog_initializer::get_view_db_charset ());
+  ADD_VIEW_DEFINITION (CTV_SYNONYM_NAME, system_catalog_initializer::get_view_synonym ());
+  ADD_VIEW_DEFINITION (CTV_DB_SERVER_NAME, system_catalog_initializer::get_view_db_server ());
+}
+
+int
+catcls_install (void)
+{
+  int error_code = NO_ERROR;
+
+  const size_t num_classes = clist.size ();
+  const size_t num_vclasses = vclist.size ();
+  std::vector<MOP> class_mop (num_classes, nullptr);
+  int save;
+  size_t i;
+  AU_DISABLE (save);
+
+  using catalog_builder = cubschema::system_catalog_builder;
+
+  for (i = 0; i < num_classes; i++)
```

Comment:
If you see the next loops below, create_and_mark_system_class () should be called in an iteration when a system view is built. If this order is not followed, assertion occurs.

Reply:
Please, explain more? Does the following code have any problem?

```
  for (i = 0; i < num_classes; i++)
      class_mop[i] = catalog_builder::create_and_mark_system_class (clist[i].name);
      if (class_mop[i] == nullptr)
	{
	  assert (false);
	  error_code = er_errid ();
	  goto end;
	}
 
      // new routine
      error_code = catalog_builder::build_class (class_mop[i], clist[i].definition);
      if (error_code != NO_ERROR)
	{
	  assert (false);
	  error_code = er_errid ();
	  goto end;
	}
    }
```

Reply:
When I build with the above code, `error_code = catalog_builder::build_class (class_mop[i], clist[i].definition);` returns -244 and assert occurs in `cubrid createdb`

Reply:
There remain questions about the cause of the problem, but since this is a comment from a readability perspective, there are no further comments. It is your choice to reflect our opinion through further analysis or leave it as is.